### ISSUE TITLE
when I click on Sort by property of a field, TE3 crashes 

### ISSUE THREAD

**Issue by paolopedron** [TabularEditor/TabularEditor3#868]
**Describe the issue**
In most of the tables of any data model, the click on "Sort By Column" property causes TE3 to crash. 

** Which version are you currently using? **
Tabular Editor 3, version: Build 3.5.1.1425
Tabular Editor 3 edition: Enterprise

**To Reproduce**
Are you connected to SSAS/AAS/Power BI or working offline?
Connected 

1. Open a model
2. Go to a table
3. Select a field
4. Go to Properties 
5. Select the "Sort By Column" property
6. Try to place the cursor in the text box or to select a field with the drop-down arrow
7. TE3 crashes

**Expected behavior**
I expect to see a list of fields to choose from. 

**Screenshots**
![image](https://user-images.githubusercontent.com/129518899/229128627-b46d019f-7d44-4822-aac2-133042dd42d8.png)

![image](https://user-images.githubusercontent.com/129518899/229128908-846e69f0-b410-4c3a-9357-4dd01dd0b94a.png)

**Desktop (please complete the following information):**
 - Windows version: Windows Server 2022 Datacenter Azure Edition

**Additional context**

**Comment by DBojsen** [comment]
Hi @paolopedron 

We are unable to reproduce right now - could I ask you to answer below questions.

Does it happen regardless of which model you load, and which field you're working on?
Does the fields in question contain special characters?
Do you see the same issue if attempting from a different server/development machine?

As a workaround for now, you can mark the column and run this C# script:
Selected.Column.SortByColumn = SelectColumn(Selected.Table.Columns);

BR David - TE3 support

**Comment by paolopedron** [comment]
Hi David,
it happens regardless of which model, and regardless of the field, but it happens on most of the fields, not 100% of them, but for example:
***@***.***


This happens
***@***.***

I don't see any special characters, it happens with various models, either dev or production.

Paolo Pedron
Group Business Intelligence Developer

Habasit International S.r.l.
via del Lavoro, 50
31016 Cordignano TV
Italy

Mobile: +39 349 939 3322
Email: ***@***.******@***.***>

[Habasit]<http://www.habasit.com/en/index.htm>

**Comment by otykier** [comment]
Hi @paolopedron 
GitHub doesn't support attaching images through e-mail, so if you attempted to add some screenshot to your previous reply, we can't see them. Could you go to the issue on GitHub: https://github.com/TabularEditor/TabularEditor3/issues/868 and upload the screenshots there, please? Thank you.

**Comment by paolopedron** [comment]
Please see the screenshots
![pic1](https://user-images.githubusercontent.com/129518899/229429025-8e87cce4-558a-4c98-b4ce-ccbb0d057f44.png)
![pic2](https://user-images.githubusercontent.com/129518899/229429039-090a8850-02e8-46ea-b6c4-2ac726da4e35.png)

**Comment by otykier** [comment]
Thanks - we're looking into it. For now, please use the workaround described above, if you need to change the Sort By Column property.

**Comment by DBojsen** [comment]
Hi @paolopedron 

We still have not been able to reproduce.
Have you had a chance to attempt the same change from a different windows machine?

BR David - TE3 Support

**Comment by paolopedron** [comment]
Hi,
unfortunately I cannot try on a different machine, as we have only one virtual machine we access.

Paolo Pedron
Group Business Intelligence Developer

Habasit International S.r.l.
via del Lavoro, 50
31016 Cordignano TV
Italy

Mobile: +39 349 939 3322
Email: ***@***.******@***.***>

[Habasit]<http://www.habasit.com/en/index.htm>

**Comment by edhans** [comment]
I have 3.6.0 and it is not crashing when I set the sort by column property, but I cannot save to the workspace after making the change. I get this:
![image](https://user-images.githubusercontent.com/45298358/235809009-37d0ad99-32bd-4774-8996-937fd4e1ee89.png)
The workaround script above doesn't help as it just sets the property. Same error when saving the model to the Premium workspace.

**Comment by edhans** [comment]
I get the same error with TE 2.18.0.
![image](https://user-images.githubusercontent.com/45298358/235810371-4fc7f373-3bde-4123-974a-1e09be21ca52.png)

Is that column ID some sort of ID that isn't being properly associated with the Sort By column? The lineage tag is a standard GUID. 

{10 minutes later}... Ok, figured it out. The "IsAvailableInMDX" was set to false. Set it to True on the Sort By field, and was able to set it. 

The UI should help out here. Not sure how... 
When I tried to set IsAvailableInMDX for Sort By back to False, I got this error:
![image](https://user-images.githubusercontent.com/45298358/235810793-37775fed-ef65-4c8b-9776-33ee1702bc48.png)

It is just 2 records, so immaterial for performance, but BPA told me to change it.

I think at the very least, invalid fields (those with IsAvailableInMDX set to false should not show up in the Sort By field dropdown, but that just prevents the error. Doesn't explain why the field isn't there. Maybe add more text down here (yellow area) to explain that? Still somewhat hidden, but it would prevent guessing or thinking there was a bug.

"If you don't see the field in the dropdown for the Sort By Column property, make sure the IsAvailableInMDX property for the desired Sort By column is set to TRUE" - that assumes you hide the fields that aren't valid for Sort By in that dropdown.
